tests_metricflow/integration/test_cases/itest_granularity.yaml

```diff
@@ -310,3 +310,142 @@ integration_test:
+  check_query: |
+    SELECT
+      metric_time__martian_day
+      , trailing_2_months_revenue
+    FROM (
+      SELECT
+        metric_time__martian_day
+        , AVG(txn_revenue) OVER (PARTITION BY metric_time__martian_day) AS trailing_2_months_revenue
+      FROM (
+        SELECT
+          ts2.martian_day AS metric_time__martian_day
+          , ts.ds AS metric_time__day
+          , SUM(r.revenue) AS txn_revenue
+        FROM {{ source_schema }}.mf_time_spine ts
+        INNER JOIN {{ source_schema }}.fct_revenue r
+          ON ({{ render_date_trunc("r.created_at", TimeGranularity.DAY) }} <= ts.ds)
+          AND ({{ render_date_trunc("r.created_at", TimeGranularity.DAY) }} > {{ render_date_sub("ts", "ds", 2, TimeGranularity.MONTH) }})
+        LEFT OUTER JOIN {{ source_schema }}.mf_time_spine ts2
+          ON ts.ds = ts2.ds
+        GROUP BY ts2.martian_day, ts.ds
+      ) ts
+    ) subq_9
+    GROUP BY metric_time__martian_day, trailing_2_months_revenue
+---
+integration_test:
+  name: derived_offset_metric_with_custom_granularity
+  description: A derived metric queried with a custom granularity
+  model: SIMPLE_MODEL
+  metrics: [ "booking_fees_last_week_per_booker_this_week"]
+  group_bys: ["metric_time__martian_day"]
+  check_query: |
+    SELECT
+      metric_time__martian_day
+      , booking_value * 0.05 / bookers AS booking_fees_last_week_per_booker_this_week
+    FROM (
+      SELECT
+        COALESCE(subq_8.metric_time__martian_day, subq_14.metric_time__martian_day) AS metric_time__martian_day
+        , MAX(subq_8.booking_value) AS booking_value
+        , MAX(subq_14.bookers) AS bookers
+      FROM (
+        SELECT
+          ts2.martian_day AS metric_time__martian_day
+          , SUM(b.booking_value) AS booking_value
+        FROM {{ source_schema }}.mf_time_spine ts
+        INNER JOIN {{ source_schema }}.fct_bookings b
+          ON {{ render_date_sub("ts", "ds", 1, TimeGranularity.WEEK) }} = {{ render_date_trunc("b.ds", TimeGranularity.DAY) }}
+        LEFT OUTER JOIN {{ source_schema }}.mf_time_spine ts2 ON ts.ds = ts2.ds
+        GROUP BY ts2.martian_day
+      ) subq_8
+      FULL OUTER JOIN (
+        SELECT
+          ts3.martian_day AS metric_time__martian_day
+          , COUNT(DISTINCT b.guest_id) AS bookers
+        FROM {{ source_schema }}.fct_bookings b
+        LEFT OUTER JOIN {{ source_schema }}.mf_time_spine ts3
+          ON {{ render_date_trunc("b.ds", TimeGranularity.DAY) }} = ts3.ds
+        GROUP BY ts3.martian_day
+      ) subq_14
+      ON subq_8.metric_time__martian_day = subq_14.metric_time__martian_day
+      GROUP BY COALESCE(subq_8.metric_time__martian_day, subq_14.metric_time__martian_day)
+    ) subq_15
+---
+integration_test:
+  name: multiple_metrics_with_custom_granularity
+  description: A derived metric queried with a custom granularity
+  model: SIMPLE_MODEL
+  metrics: [ "bookings", "listings"]
+  group_bys: ["metric_time__martian_day", "listing__ds__month"]
```

Comment:
This is exercising the new datepart logic in the linker for custom granularities, right?  (I'm just trying to confirm the things I'm learning.)

Reply:
> the new datepart logic in the linker

Maybe! What do you mean by this?

Reply:
hahaha uh oh sounds like I'm learning less than I thought.

I *think* this query exercises the changes in #1415 - we're able to groupBy DatePart on a derived metric with custom granularity because because of that PR, right?

Reply:
Ohh ok I get what you meant by `the new datepart logic in the linker` now!

> we're able to groupBy DatePart on a derived metric with custom granularity because because of that PR


Not quite. Date part and custom granularity can't be used together. That PR does 2 things:
- Bug fix: updates the `LinkableSpecResolver` to allow non-default granularity for cumulative metrics. By non-default I don't mean custom. The _default_ granularity for a dimension is the one it's defined at in the YAML definition. For example, if a metric is defined with `DAY` granularity, the default is `DAY` and all other standard granularities (`MONTH`, `YEAR`, etc.) _and_ custom granularities are considered non-default. Cumulative metrics can be queried with all granularities, which is a change we made this year, and the `LinkableSpecResolver` was out of date for that logic. Cumulative metrics still cannot be queried with date part, though. This change was all in the first commit.
- Adds custom granularities to the `LinkableSpecResolver` for all metrics. This change is in all the later commits.
There are a lot of different metric types and group by options to grasp all at once here 😅 So let me know if this makes sense or if you have questions about any of it!

As for this test - it is testing that we can use a custom granularity with multiple simple metrics, and that does rely on the logic from that PR!

Also, I realized just now that I didn't update the descriptions for these tests properly (good old copy/paste) and that probably contributed to the confusion. I'll update those before merging.

Reply:
ok, i think it does make sense.  I may need a reminder or two about some of these details in the future, but I'm definitely putting things together a bit more as I go. :)

Thank you, Courtney!